agent/options.go

```diff
@@ -0,0 +1,180 @@
+//  Copyright (c) 2018 Cisco and/or its affiliates.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at:
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package agent
+
+import (
+	"errors"
+	"reflect"
+	"time"
+
+	"github.com/ligato/cn-infra/core"
+	"github.com/ligato/cn-infra/logging/logrus"
+)
+
+type Options struct {
+	Version        string
+	MaxStartupTime time.Duration
+
+	Plugins []core.PluginNamed
+}
+
+func newOptions(opts ...Option) Options {
+	opt := Options{
+		Version:        "dev",
```

Comment:
Lets put "dev" in a constant.

Reply:
It's used in a single place and it's there just to not be empty when not set. Actually we might keep the old way of having global vars in `version.go` file which are set by linker (`-ldflags`). I put it there as an example mainly.